error_113_test.go

```diff
@@ -0,0 +1,104 @@
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestErrorUnwrap(t *testing.T) {
+	t.Run("Trivial", func(t *testing.T) {
+		err := testType.NewWithNoMessage()
+		unwrapped := errors.Unwrap(err)
+		require.Nil(t, unwrapped)
+	})
+
+	t.Run("Wrap", func(t *testing.T) {
+		err := testTypeBar1.Wrap(testType.NewWithNoMessage(), "")
+		unwrapped := errors.Unwrap(err)
+		require.Nil(t, unwrapped)
+	})
+
+	t.Run("WrapForeign", func(t *testing.T) {
+		err := testTypeBar1.Wrap(io.EOF, "")
+		unwrapped := errors.Unwrap(err)
+		require.Nil(t, unwrapped)
+	})
+
+	t.Run("Decorate", func(t *testing.T) {
+		err := Decorate(testType.NewWithNoMessage(), "")
+		unwrapped := errors.Unwrap(err)
+		require.NotNil(t, unwrapped)
+		require.True(t, IsOfType(unwrapped, testType))
+	})
+
+	t.Run("DecorateForeign", func(t *testing.T) {
+		err := Decorate(io.EOF, "")
+		unwrapped := errors.Unwrap(err)
+		require.NotNil(t, unwrapped)
+		require.True(t, errors.Is(unwrapped, io.EOF))
+	})
+
+	t.Run("Nested", func(t *testing.T) {
+		err := Decorate(Decorate(testType.NewWithNoMessage(), ""), "")
+		unwrapped := errors.Unwrap(err)
+		require.NotNil(t, unwrapped)
+		unwrapped = errors.Unwrap(unwrapped)
+		require.NotNil(t, unwrapped)
+		require.True(t, IsOfType(unwrapped, testType))
+	})
+
+	t.Run("NestedWrapped", func(t *testing.T) {
+		err := Decorate(testTypeBar1.Wrap(testType.NewWithNoMessage(), ""), "")
+		unwrapped := errors.Unwrap(err)
+		require.NotNil(t, unwrapped)
+		require.True(t, IsOfType(unwrapped, testTypeBar1))
+		unwrapped = errors.Unwrap(unwrapped)
+		require.Nil(t, unwrapped)
+	})
+
+	t.Run("NestedForeign", func(t *testing.T) {
+		err := Decorate(Decorate(io.EOF, ""), "")
+		unwrapped := errors.Unwrap(err)
+		require.NotNil(t, unwrapped)
+		unwrapped = errors.Unwrap(unwrapped)
+		require.NotNil(t, unwrapped)
+		require.True(t, errors.Is(unwrapped, io.EOF))
+	})
+}
+
+func TestErrorIs(t *testing.T) {
+	t.Run("Trivial", func(t *testing.T) {
+		err := testType.NewWithNoMessage()
+		require.True(t, errors.Is(err, testType.NewWithNoMessage()))
+		require.False(t, errors.Is(err, testTypeBar1.NewWithNoMessage()))
+	})
+
+	t.Run("Wrap", func(t *testing.T) {
+		err := testTypeBar1.Wrap(testType.NewWithNoMessage(),"")
+		require.False(t, errors.Is(err, testType.NewWithNoMessage()))
+		require.True(t, errors.Is(err, testTypeBar1.NewWithNoMessage()))
+	})
+
+	t.Run("Supertype", func(t *testing.T) {
+		err := testSubtype0.Wrap(testTypeBar1.NewWithNoMessage(),"")
+		require.True(t, errors.Is(err, testType.NewWithNoMessage()))
+		require.True(t, errors.Is(err, testSubtype0.NewWithNoMessage()))
+		require.False(t, errors.Is(err, testTypeBar1.NewWithNoMessage()))
+	})
+
+	t.Run("Decorate", func(t *testing.T) {
+		err := Decorate(testType.NewWithNoMessage(),"")
+		require.True(t, errors.Is(err, testType.NewWithNoMessage()))
+	})
+
+	t.Run("DecorateForeign", func(t *testing.T) {
+		err := Decorate(io.EOF,"")
+		require.True(t, errors.Is(err, io.EOF))
+	})
```

Comment:
We should add a test that `errorx.IsOfType(err, errorType)` penetrates `fmt.Errorf("blabla: %w", errorType.NewWithNoMessage())`.

Reply:
As of this PR, it doesn't. I intend to add both feature and tests in a separate PR.